esmtools/carbon.py

```diff
@@ -57,10 +59,57 @@ def schmidt(t):
 
     Reference
     --------
-    Sarmiento and Gruber (2006). Ocean Biogeochemical Dynamics. 
+    Sarmiento and Gruber (2006). Ocean Biogeochemical Dynamics.
     Table 3.3.1
     """
     c = [2073.1, 125.62, 3.6276, 0.043219]
     t = np.mean(t)
-    Sc = c[0] - c[1]*t + c[2]*(t**2) - c[3]*(t**3)
+    Sc = c[0] - c[1] * t + c[2] * (t ** 2) - c[3] * (t ** 3)
     return Sc
+
+
+def temp_decomp_takahashi(ds, time_dim='time', temperature='tos', pco2='spco2'):
+    """
+    Decompose spco2 into thermal and non-thermal component.
+
+    Reference
+    ---------
+    Takahashi, Taro, Stewart C. Sutherland, Colm Sweeney, Alain Poisson, Nicolas
+        Metzl, Bronte Tilbrook, Nicolas Bates, et al. “Global Sea–Air CO2 Flux
+        Based on Climatological Surface Ocean PCO2, and Seasonal Biological and
+        Temperature Effects.” Deep Sea Research Part II: Topical Studies in
+        Oceanography, The Southern Ocean I: Climatic Changes in the Cycle of
+        Carbon in the Southern Ocean, 49, no. 9 (January 1,2002): 1601–22.
+        https://doi.org/10/dmk4f2.
+
+    Input
+    -----
+    ds : xr.Dataset containing spco2[ppm] and tos[C or K]
+
+    Output
+    ------
+    thermal, non_thermal : xr.DataArray
+        thermal and non-thermal components in ppm units
+
+    """
+    fac = 0.0432
+    tos_mean = ds[temperature].mean(time_dim)
+    tos_diff = ds[temperature] - tos_mean
+    thermal = ds[temperature].mean(time_dim) * (np.exp(tos_diff * fac))
+    non_thermal = ds[pco2] * (np.exp(tos_diff * -fac))
+    return thermal, non_thermal
+
+
+def potential_pco2(t_insitu, pco2_insitu):
+    """
+    Calculate potential pco2 in the inner ocean. Requires the first index of
+    depth to be at the surface.
+
```

Comment:
Could you add the input descriptions here? I assume `t_insitu` is raw temperature with depth. `pco2_insitu` is pCO2 with depth? Looks good otherwise